sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -193,9 +198,62 @@ public Mono<CosmosItemResponse<Object>> deleteItem(String itemId,
 
+        this.encryptionProcessor.initEncryptionSettingsIfNotInitializedAsync();
+        EncryptionSettings encryptionSettings = this.encryptionProcessor.getEncryptionSettings();
+
+        try {
+            itemId = checkAndGetEncryptedId(itemId, encryptionSettings);
+            partitionKey = checkAndGetEncryptedPartitionKey(partitionKey, encryptionSettings);
+        } catch (JsonProcessingException | MicrosoftDataEncryptionException e) {
+            return Mono.error(e);
+        }
         return container.deleteItem(itemId, partitionKey, requestOptions);
     }
 
+    private String checkAndGetEncryptedId(String itemId, EncryptionSettings encryptionSettings) throws MicrosoftDataEncryptionException {
+        if (this.encryptionProcessor.getClientEncryptionPolicy().getIncludedPaths().stream().anyMatch(includedPath -> includedPath.getPath().substring(1).equals(Constants.PROPERTY_NAME_ID))) {
+            return itemId;
+        }
+        return this.encryptionProcessor.encryptAndSerializeValue(encryptionSettings, itemId, Constants.PROPERTY_NAME_ID);
+    }
+
+    private PartitionKey checkAndGetEncryptedPartitionKey(PartitionKey partitionKey, EncryptionSettings encryptionSettings) throws JsonProcessingException, MicrosoftDataEncryptionException {
+        if (encryptionSettings.getPartitionKeyPaths().isEmpty() || partitionKey == null) {
+            return partitionKey;
+        }
+        JsonNode partitionKeyNode = EncryptionUtils.getSimpleObjectMapper().readTree(partitionKey.toString());
+        if (partitionKeyNode.isArray()) {
+            ArrayNode arrayNode = (ArrayNode) partitionKeyNode;
+            PartitionKeyBuilder partitionKeyBuilder = new PartitionKeyBuilder();
+
+            for (String path : encryptionSettings.getPartitionKeyPaths()) {
+                // case: partition key path is /a/b/c and the client encryption policy has /a in path.
+                // hence encrypt the partition key value with using its top level path /a since /c would have been encrypted in the document using /a's policy.
+                String partitionKeyPath = path.split("/")[0];
+
+                String childPartitionKey = arrayNode.elements().next().toString();
+                if (this.encryptionProcessor.getClientEncryptionPolicy().getIncludedPaths().stream().anyMatch(includedPath -> includedPath.getPath().substring(1).equals(partitionKeyPath))) {
+                    partitionKeyBuilder.add(childPartitionKey);
+                    continue;
+                }
+                partitionKeyBuilder.add(this.encryptionProcessor.encryptAndSerializeValue(encryptionSettings, childPartitionKey, partitionKeyPath));
+            }
+            return partitionKeyBuilder.build();
+        }
+
+        else {
+            if (encryptionSettings.getPartitionKeyPaths().size() > 1) {
+                throw new MicrosoftDataEncryptionException("There should only be 1 PartitionKeyPath.");
```

Comment:
Is that a temporary limitation or going to be a permanent restriction for end-to-end encryption (the decision to not support MultiHash PartitionKeyDefinition). If so, is the doc calling this limitation out clearly for both features (encryption but also multiHash partitioning - so that customers get the heads-up)

Reply:
I have logic for handling MultiHash PartitionKeyDefinition, from lines 227 - 247.